src/pystack/__main__.py

```diff
@@ -357,9 +357,24 @@ def process_core(parser: argparse.ArgumentParser, args: argparse.Namespace) -> N
         print(format_failureinfo_information(corefile_analyzer.extract_failure_info()))
 
     if not is_elf(executable):
-        raise errors.InvalidExecutable(
-            f"The provided executable ({executable}) doesn't have an executable format"
+        first_map = next(
+            (map for map in corefile_analyzer.extract_maps() if map.path is not None),
+            None,
         )
+        if (
+            first_map is not None
+            and first_map.path is not None
```

Comment:
This condition is unnecessary, right? It's already checked in the genexpr passed to next()

Reply:
Yeah, but is so mypy doesn't complain on the next call